### PR TITLE
shell: Use PF's about modal

### LINKED ISSUE #15961: Adopt PatternFly's about modal widget
The shell currently has an about modal that looks like this:

image

But [PatternFly has an about modal](https://www.patternfly.org/v4/components/about-modal) that looks like:

image

It supports different backgrounds too, so perhaps I should cook something up...

image

Ideally, we might want to list the versions of not just the main Cockpit version, but various additional Cockpit-related packages that are also installed, like cockpit-machines, cockpit-podman, cockpit-ostree, etc.

### LINKED ISSUE #15962: shell: Show versions of cockpit-* packages
In the about modal, we should list not just the main Cockpit version, but various additional Cockpit-related packages that are also installed, like cockpit-machines, cockpit-podman, cockpit-ostree, etc.

### PR DESCRIPTION
**Pull request by marusak**
## shell: Show package version in 'About web console' modal

Screenshot from 2021-08-30 14-07-52about

### CODE REVIEW

**Comment by garrett** [comment]
Great so far. Looking forward to the various cockpit-* version support!

**Comment by marusak** [comment]
With versions:
Screenshot from 2021-08-30 14-07-52about

File: pkg/shell/shell-modals.jsx [hunk 1]
```diff
@@ -20,39 +20,85 @@
...
+                productName={_("Web Console")}
+                brandImageSrc="../shell/images/cockpit-icon.svg"
+                brandImageAlt={_("Web console logo")}
+                backgroundImageSrc="../shell/images/bg-plain.jpg"
+            >
+                <div>{_("Cockpit is an interactive Linux server admin interface.")}</div>
+                <div><a rel="noopener noreferrer" target="_blank" href="https://cockpit-project.org/">{_("Project website")}</a></div>
+                <TextContent>
+                    <TextList component="dl">
+                        {this.state.packages === null && <span>{_("Loading packages...")}</span>}
```
Comment (KKoukiou) [inline pkg/shell/shell-modals.jsx:91]: At first sight I thought this in invalid - but it seems correct HTML https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl

**Review by KKoukiou — DISMISSED** [review]
I like the new look! The package manager detection logic is a copy paste from pkg/shell/shell-modals.jsx so it could be moved to pkg/lib if you want. 

The component could remain functional and a hook can be used for the function to run after mounting. 

All comments are optional! 

Super nice, thanks :)

**Comment by marusak** [comment]
> Of course tests - just noticed they fail.

Main is broken, unrelated to this PR. But of course lets wait until main is fixed.

> You did not need to rewrite this to class component, as it can be done nicely with react hooks

I haven't yet fully come to terms with hooks :) Is there actual advantage for using this or just a bit simpler syntax?

**Comment by martinpitt** [comment]
> Main is broken

the LUKS testReboot fix landed yesterday. There are still some big flaky tests (Marius is at them), but in general they should be able to go green.

**Comment by KKoukiou** [comment]
> > Of course tests - just noticed they fail.
> 
> Main is broken, unrelated to this PR. But of course lets wait until main is fixed.
> 
> > You did not need to rewrite this to class component, as it can be done nicely with react hooks
> 
> I haven't yet fully come to terms with hooks :) Is there actual advantage for using this or just a bit simpler syntax?

The latter. And the diff is nicer when you already had a functional component before.

**Comment by marusak** [comment]
> the LUKS testReboot fix landed yesterday.

rebased to get a clean run then

**Review by KKoukiou — DISMISSED** [review]

**Comment by marusak** [comment]
ah, `fedora-coreos` does not have bridge

**Review by KKoukiou — APPROVED** [review]

**Pull request merged**